Added event flow to Perfetto output for hipEventRecord and hipEventSynchronize

ajanicijamd opened the pull request:
for SWDEV-499010

ajanicijamd committed:
Added event flow to Perfetto output for hipEventRecord and hipEventSynchronize

ajanicijamd committed:
Fixed formatting

ajanicijamd committed:
Fixed formatting

dgaliffiAMD commented on source/lib/rocprof-sys/library/rocprofiler-sdk.cpp:
```diff
@@ -827,6 +827,121 @@ tool_tracing_buffered(rocprofiler_context_id_t /*context*/,
+                                                                     _corr_id);
+                                }
+                            });
+                        tracing::pop_perfetto(category::rocm_memory_copy{}, "", _track,
```
```suggestion
                        tracing::pop_perfetto(category::rocm_hip_api{}, "", _track,
```
Looks like a typo?

ajanicijamd replied:
Absolutely a typo - a blind copy-paste :-)

dgaliffiAMD commented on source/lib/rocprof-sys/library/rocprofiler-sdk.cpp:
```diff
@@ -827,6 +827,121 @@ tool_tracing_buffered(rocprofiler_context_id_t /*context*/,
+                                                                     _corr_id);
+                                }
+                            });
+                        tracing::pop_perfetto(category::rocm_memory_copy{}, "", _track,
```
```suggestion
                        tracing::pop_perfetto(category::rocm_hip_api{}, "", _track,
```

dgaliffiAMD commented on source/lib/rocprof-sys/library/rocprofiler-sdk.cpp:
```diff
@@ -827,6 +827,121 @@ tool_tracing_buffered(rocprofiler_context_id_t /*context*/,
+                auto* record = static_cast<rocprofiler_buffer_tracing_hip_api_record_t*>(
+                    header->payload);
+
+                if(record->operation == ROCPROFILER_HIP_RUNTIME_API_ID_hipEventRecord)
```
```suggestion
                const auto _ops = std::unordered_set<int32_t>{
                    ROCPROFILER_HIP_RUNTIME_API_ID_hipEventRecord,
                    ROCPROFILER_HIP_RUNTIME_API_ID_hipEventSynchronize,
                };
                
                if(_ops.count(record->operation) > 0)
```

It looks like both handlers are identical. Can you do something like this and then remove the "else-if" block?
Are there other operations that we want to handle?

dgaliffiAMD commented on source/lib/rocprof-sys/library/rocprofiler-sdk.cpp:
```diff
@@ -827,6 +827,121 @@ tool_tracing_buffered(rocprofiler_context_id_t /*context*/,
+                    {
+                        auto _track_desc = [](rocprofiler_thread_id_t _tid) {
+                            const auto& _tid_v = thread_info::get(_tid, SystemTID);
+                            return JOIN("", "HIP Runtime API Thread ",
```
```suggestion
                            return JOIN("", "GPU HIP Runtime Thread ",
```
Suggesting the "GPU" prefix, so it is grouped with the other domains.

ajanicijamd committed:
Update source/lib/rocprof-sys/library/rocprofiler-sdk.cpp

I will have to test this change. Previously I had just one if-branch, but then a very mysterious thing was happening - no event flow. So I kept the two if-branches separate. I know, that is against any logic. I'll accept the suggestion and run the test again.

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

ajanicijamd committed:
Update source/lib/rocprof-sys/library/rocprofiler-sdk.cpp

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

ajanicijamd committed:
Update source/lib/rocprof-sys/library/rocprofiler-sdk.cpp

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

ajanicijamd committed:
Update source/lib/rocprof-sys/library/rocprofiler-sdk.cpp

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

dgaliffiAMD committed:
Format source

Signed-off-by: David Galiffi <David.Galiffi@amd.com>

dgaliffiAMD commented:
Superseded by https://github.com/ROCm/rocprofiler-systems/pull/296
